src/box/gc.c

```diff
@@ -1103,6 +1104,26 @@ gc_load_consumers(void)
 		consumer->is_persistent = true;
 		/* Unref WAL GC delay for each replica having a consumer. */
 		gc_delay_unref();
+
+		/* The following code relies on having one type of consumers. */
+		assert(consumer->type == GC_CONSUMER_REPLICA);
+		struct replica *replica = replica_by_uuid(&consumer->uuid);
+		if (replica != NULL) {
+			/*
+			 * Since the function is called on configuration, only
+			 * regular replicas can appear here.
+			 */
+			assert(!replica->anon);
+			continue;
+		}
+		/*
+		 * Create an anonymous replica object and register its
+		 * consumer so that it won't be lost.
+		 */
+		replica = replicaset_add_anon(&consumer->uuid);
+		replica->gc = gc_consumer_register(&consumer->vclock,
```

Comment:
Introduced a new function `replication_anon_gc_init` that is called after `gc_load_consumers`. By the way, added `replication_anon_gc_fiber != NULL` checks to `replication_shutdown` and `box_set_replication_anon_ttl` since the fiber is started later.